roles/gitcontrol/files/github_api.py

```diff
@@ -7,21 +7,39 @@
 
 github_user = os.getenv('GITHUB_USER')
 github_token = os.getenv('GITHUB_TOKEN')
-headers = {'Authorization': f'token {github_token}'}
-bad_statuses = [400, 404]
+headers = {
+    'Authorization': f'token {github_token}',
+    'Accept': 'application/vnd.github.luke-cage-preview+json'
+}
+bad_statuses = [400, 403, 404, 422]
 
 
 def read_yaml_file(path, org, endpoint, repo_name=None):
-    if endpoint == 'manage_collaborators':
+    if endpoint in ['manage_collaborators', 'branch_protection', 'options']:
         path += f'/{org}/repositories/{repo_name}.yml'
     with open(path, 'r') as file:
-        data = yaml.load(file)
+        data = yaml.load(file, Loader=yaml.FullLoader)
     return data
 
 
+def update_options(github_api, owner, repo_name, repo):
+    output = ''
+    options = repo[repo_name]
+    options.update({'name': repo_name})
+    res = requests.patch(
+        f'{github_api}/repos/{owner}/{repo_name}',
+        json=options,
+        headers=headers,
+        timeout=15)
+    if res.status_code in bad_statuses:
+        output += f'options change not applied: {res.status_code}, error is: {res.text}\n'
+    return print(output)
+
+
 def manage_collaborators(github_api, owner, repo_name, repo):
     output = ''
     collaborators = repo[repo_name]['collaborators']
+    teams = repo[repo_name]['teams']
     for clb in collaborators:
```

Comment:
You can use `key, val in collaborators.items()` to iterate through both keys an values